sdk/log/processor.go

```diff
@@ -26,6 +26,24 @@ type Processor interface {
 	// Before modifying a Record, the implementation must use Record.Clone
 	// to create a copy that shares no state with the original.
 	OnEmit(ctx context.Context, record Record) error
+	// Enabled returns whether the Processor will process for the given context
+	// and record.
+	//
+	// The passed record is likely to be a partial record with only the
+	// bridge-relevant information being provided (e.g a record with only the
+	// Severity set). If a Logger needs more information than is provided, it
+	// is said to be in an indeterminate state (see below).
+	//
+	// The returned value will be true when the Processor will process for the
+	// provided context and record, and will be false if the Processor will not
+	// process. The returned value may be true or false in an indeterminate
+	// state. An implementation should default to returning true for an
+	// indeterminate state, but may return false if valid reasons in particular
+	// circumstances exist (e.g. performance, correctness).
+	//
+	// Before modifying a Record, the implementation must use Record.Clone
+	// to create a copy that shares no state with the original.
+	Enabled(ctx context.Context, record Record) bool
```

Comment:
Should we update the docs for `OnEmit` and document that implementations should not process when `Enabled` would return `false` for the provided context and record?

Reply:
This seems self-evident given the meaning of `Enabled` and its documentation. Do you think adding that to the already long documentation of `OnEmit` might muddy the waters more than needed?

Reply:
https://github.com/open-telemetry/opentelemetry-go/pull/5071/commits/76c8fb61719fe4db42fcfb1687dbbedf22c4e196